Sources/SwiftSyntaxBuilderGeneration/Templates/ResultBuildersFile.swift

```diff
@@ -0,0 +1,355 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+// TODO: Move this to correct file
+let SYNTAX_NODES: [Node] = COMMON_NODES +
+EXPR_NODES +
+DECL_NODES +
+ATTRIBUTE_NODES +
+STMT_NODES +
+GENERIC_NODES +
+TYPE_NODES +
+PATTERN_NODES +
+AVAILABILITY_NODES
+
+let resultBuildersFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    StructDecl(attributes: TokenSyntax.identifier("@resultBuilder").withLeadingTrivia(.newlines(1)).withTrailingTrivia(.newlines(1)),
+               modifiers: TokenSyntax.public,
+               identifier: node.syntaxKind + "Builder") {
+
+        TypealiasDecl(leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// The type of individual statement expressions in the transformed function,"),
+            .newlines(1),
+            .docLineComment("/// which defaults to Component if buildExpression() is not provided."),
+            .newlines(1)
+        ],
+                      attributes: nil,
+                      modifiers: [TokenSyntax.public],
+                      identifier: TokenSyntax.identifier("Expression"),
+                      initializer: TypeInitializerClause(value: node.expressibleAs))
```

Comment:
What do you think about indenting these (and the other) decls as

```suggestion
        TypealiasDecl(
          leadingTrivia: [
            .newlines(1),
            .docLineComment("/// The type of individual statement expressions in the transformed function,"),
            .newlines(1),
            .docLineComment("/// which defaults to Component if buildExpression() is not provided."),
            .newlines(1)
          ],
          attributes: nil,
          modifiers: [TokenSyntax.public],
          identifier: TokenSyntax.identifier("Expression"),
          initializer: TypeInitializerClause(value: node.expressibleAs)
        )
```

so the indentation doesn't jump back and forth?

Reply:
The `attributes`, `modifiers` have actually the correct indentation they are under the `leadingTrivia`.
It is the array that is wrongly indented by Xcode. 

We had the same issue here: https://github.com/apple/swift-syntax/pull/469#discussion_r908564456

Properly a bug in Xcode

Reply:
We could use a syntax like 


```
        TypealiasDecl(leadingTrivia:
                        [
                          .newlines(1),
                          .docLineComment("/// The type of individual statement expressions in the transformed function,"),
                          .newlines(1),
                          .docLineComment("/// which defaults to Component if buildExpression() is not provided."),
                          .newlines(1)
                        ],
                      attributes: nil,
                      modifiers: [TokenSyntax.public],
                      identifier: TokenSyntax.identifier("Expression"),
                      initializer: TypeInitializerClause(value: node.expressibleAs))
```

Reply:
Yeah, that makes sense. Though I think putting the parameters on a new indentation level would still benefit readability, especially with long lines (like the doc comments).

Reply:
I saw you did the same in `TokensFile`. 
I think we should do the same, so will change 💪

Reply:
Something like this @fwcd ?

Reply:
Yes, that looks great!